src/App.tsx

```diff
@@ -1,14 +1,94 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { FilterStatus } from './types/filter';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectTodos, setSelectTodos] = useState<Todo | null>(null);
+  const [filter, setFilter] = useState(FilterStatus.All);
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  const delay = 300;
+
+  const filteredTodos = useMemo(() => {
+    let filteredTodo = todos;
+
+    if (filter === FilterStatus.Active) {
+      filteredTodo = filteredTodo.filter(todo => !todo.completed);
+    }
+
+    if (filter === FilterStatus.Completed) {
+      filteredTodo = filteredTodo.filter(todo => todo.completed);
+    }
+
+    if (appliedQuery) {
+      filteredTodo = filteredTodo.filter(todo =>
+        todo.title.toLowerCase().includes(appliedQuery.toLowerCase().trim()),
+      );
+    }
+
+    return filteredTodo;
+  }, [todos, filter, appliedQuery]);
+
+  const debounce = (callback: (args: string) => void) => {
+    let timerId = 0;
+
+    return (args: string) => {
+      window.clearTimeout(timerId);
+
+      timerId = window.setTimeout(() => {
+        callback(args);
+      }, delay);
+    };
+  };
+
+  const applyDebounceQuery = useCallback(
```

Comment:
you can use `useMemo` here
or just disable this lint rule for this line

Please, create the debounce function outside of the component and remove it from the dependency array

Reply:
Not fixed 
<img width="883" alt="image" src="https://github.com/mate-academy/react_dynamic-list-of-todos/assets/64750958/22fc41d5-bd23-4907-898f-1f4f4dde5c14">